internal/juju/models.go

```diff
@@ -124,6 +126,8 @@ func (c *modelsClient) GetModelByName(name string) (*params.ModelInfo, error) {
 
 	modelInfo := results[0].Result
 
+	c.AddModel(modelInfo.Name, modelUUID, model.ModelType(modelInfo.Type))
```

Comment:
suggestion: Put the cache on a separate object so that we can do `c.ModelCache.AddMode()`.

Reply:
That's a great suggestion long term. It's not currently possible due to deadlines (code freeze in 2 days) and priorities.  When the user name is added in will be a great time to do that refactoring.